classes/Cart.php

```diff
@@ -61,6 +61,12 @@ class CartCore extends ObjectModel
     /** @var int Language ID */
     public $id_lang;
 
+    /** @var int Language ID
+     * This is the same as $id_lang except in the following case:
+     * If $id_lang is invalid (e.g. due to a removed language) $id_lang_associated is the default language
+     */
+    protected $id_lang_associated;
```

Comment:
This should be handled in ObjectModel this way all the classes will profit from the improvement

Reply:
I had a pending review that said exactly that 😅